todo_app/data/mongo_items.py

```diff
@@ -0,0 +1,144 @@
+from datetime import datetime, date
+import json
+import os
+import pymongo
+from bson.objectid import ObjectId
+
+MONGO_URI = os.getenv('MONGO_CONNECTION_STRING')
+DATABASE_NAME = os.getenv('DATABASE_NAME')
```

Comment:
Aha, your tests fail for me at this stage when I run them locally (with `poetry run pytest`) despite having set these environment variables. It looks like you dodged this potential issue back in M2/3 when it sometimes first surfaces, so apologies for the long comment explaining the likely cause!

The issue I'm seeing is related to the specific order in which certain events happen, which is why you'll dodge it in some (most!) scenarios:
* When you run the tests, they load the test file first
* That test file then imports other files (e.g. `from todo_app.data.mongo_items import Item, List`)
* At that point, this file is then read/executed from top-to-bottom
  * Functions are _defined_, but not actually run
  * However these lines execute, and try to load environment variables/connect to the DB even though `.env` hasn't yet been loaded! 😞 (e.g. in your unit tests, we don't actually need this, but it fails us anyway!)

You'll not see this in some places because:
* When starting the app with `poetry run flask run`, it loads `.env` before it executes the first Python file
* If you run the tests through VSCode's explorer window, that actually loads your `.env` silently before it runs the tests!
* If you run through Docker, then the use of `--env`/`--env-file` flags mean that the variables are set before the commands run, rather than loaded dynamically

So we can fix this by finding another way to load the environment variables first (e.g. have the tests load them before they import the file) but I'm not a fan of that personally as it leads to thinking about things in a very specific order which typically leads to flaky behaviour on changing things!

My preferred solution here then would be to handle this logic in a function that we can decide when it runs.

If we do want a global connection (which does help performance) then we want some sort of singleton pattern, the simplest lazy approach to which might look like:

```python

db = None
default_collection = None

def init_my_db():
    MONGO_URI = os.getenv('MONGO_CONNECTION_STRING')
    DATABASE_NAME = os.getenv('DATABASE_NAME')

    client = pymongo.MongoClient(MONGO_URI)
    global db, default_collection
    db = client[DATABASE_NAME]
    lists = ["to-do", "doing", "done"]
    default_collection = db['to-do']
    for lst in lists:  # Ensuring collections exist, makes them if they don't
        collection = db[lst]
        temp_item = collection.insert_one({})
        collection.delete_one({'_id': ObjectId(temp_item.inserted_id)})
```
Where `init_db` now gets called from `app.py` as part of the `create_app` function

However, this reveals some differences in your patching which we need to tweak - documented on the relevant function in `test_mocking.py`